base/threadcall.jl

```diff
@@ -43,7 +43,9 @@ macro threadcall(f, rettype, argtypes, argvals...)
         push!(body, :(p += Core.sizeof($T)))
         push!(args, arg)
     end
+    push!(body, :(gc_state = ccall(:jl_gc_safe_enter, Int8, (), )))
```

Comment:
So a problem with this is that the `ccall` lowering will insert calls to `cconvert` and then `unsafe_convert`. `cconvert` is designed to allocate. 

So you manually need to insert the code to lower include a`GC.@preserve`

Reply:
See https://github.com/JuliaGPU/CUDA.jl/blob/4d453eddaca1daca9ee7d812692bf2f80928c6ba/lib/utils/call.jl#L226

Reply:
Have I told you how much I dislike threadcall?

Reply:
Also see https://github.com/JuliaInterop/ObjectiveC.jl/commit/2578dc2cfd98efc60b3db1ef0b559c9dc9067448. It would almost be easier to fix https://github.com/JuliaLang/julia/pull/49933 at this point...

Reply:
The issue is that we need to backport this :(